guides/source/caching_with_rails.md

```diff
@@ -370,6 +370,18 @@ There are some common options that can be used by all cache implementations. The
 
 * `:race_condition_ttl` - This option is used in conjunction with the `:expires_in` option. It will prevent race conditions when cache entries expire by preventing multiple processes from simultaneously regenerating the same entry (also known as the dog pile effect). This option sets the number of seconds that an expired entry can be reused while a new value is being regenerated. It's a good practice to set this value if you use the `:expires_in` option.
 
+When using Puma or another threaded server it is recommended to use a connection pool to prevent thread contention. Connection pooling is only supported by the `MemCacheStore` and `RedisCacheStore`. To enable the connection pool first add the `connection_pool` gem to your Gemfile.
+
+```ruby
+gem 'connection_pool'
+```
+
+Next set either the `:pool_size` and/or `:pool_timeout` options.
+
+* `:pool_size` - This options sets the number of connections per process. It is recommended to set it to the number of threads per process of the app server.
```

Comment:
> It is recommended

By whom? 😄 depending on your configuration, this could flip the failure mode from "requests get slower when cache operations are slow" to the potentially disastrous "many more cache connections are opened when cache operations are slow".

I don't have a concrete suggestion here, but notice how the Configuring Rails Applications guide avoids prescribing a particular configuration while still giving some guidance:

https://github.com/rails/rails/blob/8eef8791fab0985060e7a17687698383b3d47c2c/guides/source/configuring.md#database-pooling

> If you get the above error, you might want to increase the size of the connection pool by incrementing the `pool` option in `database.yml`